stl/inc/tuple

```diff
@@ -918,7 +918,6 @@ struct _Tuple_element {}; // backstop _Tuple_element definition
 
 template <class _This, class... _Rest>
 struct _Tuple_element<_This, tuple<_This, _Rest...>> { // select first element
-    static_assert(!_Is_any_of_v<_This, _Rest...>, "duplicate type T in get<T>(tuple)");
     using _Ttype = tuple<_This, _Rest...>;
 };

```

Comment:
This is great and correct, but we should go further. Since you're now calculating `_Idx` with `_Meta_find_unique_index`, we're paying a throughput cost and code complexity to calculate `_Ttype` with `_Tuple_element`, essentially repeating work.

Instead, if we switch over to the index domain, we can drop `_Tuple_element` entirely, and rely on the `_MSVC_KNOWN_SEMANTICS` throughput optimization for Standard `tuple_element`:

https://github.com/microsoft/STL/blob/56ad8253a8efe3bbb5bbde42c9fd755764959f44/stl/inc/utility#L644-L649